deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -154,7 +154,7 @@ class ZeroParamStatus(Enum):
 _orig_torch_empty = torch.empty
 
 
-def empty_cuda_tensor(*size, **kwargs):
+def empty_cuda_tensor_half(*size, **kwargs):
     if not 'device' in kwargs.keys():
         kwargs['device'] = torch.device('cuda:{}'.format(os.environ["LOCAL_RANK"]))
     tensor = _orig_torch_empty(*size, **kwargs)
```

Comment:
Is this missing .half()?

Reply:
The .half() happen in return

Reply:
This is interesting. I had to view the file in order to see return with half. I don't know why the diff does not show the return at all.